server/src/main/java/io/crate/expression/scalar/timestamp/CurrentTimeFunction.java

```diff
@@ -0,0 +1,111 @@
+ */
+
+package io.crate.expression.scalar.timestamp;
+
+import com.google.common.math.LongMath;
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+import io.crate.types.TimeTZ;
+
+import javax.annotation.Nullable;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public class CurrentTimeFunction extends Scalar<TimeTZ, Integer> {
+
+    public static final String NAME = "current_time";
+    public static final int DEFAULT_PRECISION = 6; // micro seconds
+
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.TIMETZ.getTypeSignature()),
+            (signature, args) -> new CurrentTimeFunction(signature));
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.TIMETZ.getTypeSignature()),
+            (signature, args) -> new CurrentTimeFunction(signature));
+    }
+
+
+    private final Signature signature;
+    private final FunctionInfo info;
+
+    public CurrentTimeFunction(Signature signature) {
+        this.signature = signature;
+        this.info = new FunctionInfo(
+            new FunctionIdent(
+                NAME,
+                signature.getArgumentTypes().isEmpty() ? List.of() : List.of(DataTypes.INTEGER)),
+            DataTypes.TIMETZ,
+            FunctionInfo.Type.SCALAR,
+            Collections.emptySet());
+    }
+
+    @Override
+    @SafeVarargs
+    public final TimeTZ evaluate(TransactionContext txnCtx, Input<Integer>... args) {
+        Integer precision = DEFAULT_PRECISION;
+        if (args.length == 1) {
+            precision = args[0].value();
+            if (precision == null) {
+                throw new IllegalArgumentException(String.format(
+                    Locale.ENGLISH, "NULL precision not supported for %s", NAME));
+            }
+        }
+        if (precision < 0 || precision > DEFAULT_PRECISION) {
+            throw new IllegalArgumentException(String.format(
+                Locale.ENGLISH, "precision must range between [0..%d]", DEFAULT_PRECISION));
+        }
+        long currentTimeMicros = txnCtx.currentTimeMicros();
+        long dateDeltaMillis = Instant.ofEpochMilli(currentTimeMicros / 1000L).truncatedTo(ChronoUnit.DAYS).toEpochMilli();
+        long microsFromUTC = currentTimeMicros - (dateDeltaMillis * 1000L);
+        long f = (long) Math.pow(10, 6 - precision);
+        microsFromUTC = LongMath.divide(microsFromUTC, f, RoundingMode.DOWN) * f;
```

Comment:
Is this the same result as `Clock.systemUTC().instant().get(ChronoField.MICRO_OF_DAY);` would return? 

If so, maybe the `TransactionContext` could be changed to cache the `instant` instead (afaik it's a singleton, so would be safe).